pgtransaction/tests/test_atomic.py

```diff
@@ -88,77 +83,58 @@ class MockError(Exception):
         pass
     with pytest.raises(MockError):
         with atomic(isolation_level="REPEATABLE READ"):
-            Trade.objects.create(company='Coca Cola', price=1)
+            ddf.G(Trade)
             raise MockError
-    assert 0 == Trade.objects.count()
+    assert not Trade.objects.exists()
 
 
 @pytest.mark.django_db(transaction=True)
 def test_pg_atomic_nested_atomic_rollback():
     with atomic(isolation_level="REPEATABLE READ"):
-        Trade.objects.create(company='Coca Cola', price=1)
+        trade = ddf.G(Trade, company='Coca Cola')
         try:
-            with atomic():
-                Trade.objects.create(company='Coca Cola 2', price=1)
-                raise RuntimeError
-        except RuntimeError:
+            with atomic():  # pragma: no branch
+                trade.id = None
+                trade.save()
+        except IntegrityError:
```

Comment:
Any particular reason we switched to IntegrityError here? Main idea of this test was to demonstrate that any exception, not just ones coming from db operations, behaved as expected with respect to nested atomics (i.e. savepoints).

Reply:
I see, will change back!

Reply:
Addressed in 4acadc0e7f3a6f368444a63c2a0187b6370c0750